### ISSUE TITLE
adblockers create errors on browser console

### ISSUE THREAD

**Issue by petitkriket** [grafana/faro-web-sdk#497]
## Description
Capturing metrics with a cloud based instance floods console with errors because of the users adblockers

## Steps to reproduce
1. Add Faro to the project
2. Send a metric to a Grafana.com cloud hosted instance using the Faro SDK
3. See a error clogged browser console 
4. Disable adblocker
5. Console is clean

## Expected behavior
1. Add Faro to the project
2. Send a metric to a Grafana managed instance using the Faro API
3. No error in the console while data is not sent and user trace is not sent
6. No error in the console while data is sent because it's whitelisted to adblockers

## Actual behavior

Console is full of errors which makes seeing actual bugs more difficult

## Environment

- **SDK version: 1.3.9 iiffe**
- **SDK instrumentations: tracing is enabled**
- **Device type:**
- **Device name:**
- **OS: MacOs**
- **Browser: Chrome 121.0.0**

## Demo

<img width="1469" alt="Screenshot 2024-02-22 at 23 33 51" src="https://github.com/grafana/faro-web-sdk/assets/1753019/38cb2be1-f1d1-4d4a-8e73-94eeb862cf22">

## Context

grafana saas cloud domain is present on [adblocker ban lists](https://github.com/search?q=repo%3Aeasylist%2Feasylist%20graFANA&type=code)
So I guess it would work fine on a self hosted Grafana but not on a managed one

**Comment by eskirk** [comment]
hi @petitkriket, thanks for the report. 

we are aware of this issue and we have some solutions in mind to help circumvent this problem - as a user of ublock myself, I really want to fix this.

I do worry this may end up being a game of cat and mouse with easylist VS faro endpoints - especially since myself and the rest of the team fully respects the user's rights to limit whatever type of payloads their browser may be sending to endpoints across the internet. 

I will check in on the progress of our solution and let you know when I know more.

**Comment by petitkriket** [comment]
thank for you swift reply ! totally agree with you on the matter of privacy. happy to know and not surprised that you're well aware of the issue, curious about the solution you and your team will pick !

the additional errors produced from the SDK creates unwanted noise IMO. a single warning would be a nice to have..

kind regards

**Comment by daniel-hayes** [comment]
@eskirk are there any updates to this? Ideally we could pause transports if the collector is being blocked by an extension, but I don't think we have the ability to catch errors coming from the SDK. I think setting up a proxy would work, but maybe also having a health check that can allow us to pause transports on load would suffice?

**Comment by eskirk** [comment]
@daniel-hayes I like where your head is at. I am going to bring this issue up in our upcoming planning meeting and see what options we have here. a health check would go a long way at reducing console noise - or even at least a more specific error showing that this request was blocked by an ad blocker. thanks for putting this back on my radar 👍🏼

**Comment by daniel-hayes** [comment]
@eskirk thanks for the update!

**Comment by dirtydiesel46** [comment]
HI @eskirk 

Any updates on this one yet ? How did the planning meeting go / has it happened yet ?

**Comment by eskirk** [comment]
@dirtydiesel46 our current planned course of action is to check for the presence of an adblocker during faro initialization and essentially pause Faro if an adblocker is detected. then there at least wont be console errors. I do not quite have an ETA, but I will put it on my list.

**Comment by abdullah-retorio** [comment]
Hello, is there any update on this? We are planning to use Faro on the production level soonish.

**Comment by eskirk** [comment]
hi all - we do not yet have the means to detect and pause transports when an adblocker is present. we do have a workaround where users are able to [proxy RUM data to the Faro cloud collector/open source collectors](https://grafana.com/docs/grafana-cloud/monitor-applications/frontend-observability/instrument/data-proxy/). 

if you need immediate unblocking for this issue, I would recommend following these docs and setting up a proxy to avoid the adblock list. hopefully this helps in the meantime while we find the time for this feature. 

we also encourage community contributions - if anyone was willing and able to make this happen we would happily review, merge, and maintain that feature.

**Comment by eskirk** [comment]
> Can Faro send the data to a Grafana Alloy instance that is self-hosted on a cloud VM?

@abdullah-retorio yes it can, [here are some recent docs for how to do so](https://grafana.com/docs/alloy/latest/reference/components/faro/faro.receiver/)
